src/tools/clippy/clippy_utils/src/qualify_min_const_fn.rs

```diff
@@ -367,10 +367,21 @@ fn is_const_fn(tcx: TyCtxt<'_>, def_id: DefId, msrv: Option<RustcVersion>) -> bo
                 // Checking MSRV is manually necessary because `rustc` has no such concept. This entire
                 // function could be removed if `rustc` provided a MSRV-aware version of `is_const_fn`.
                 // as a part of an unimplemented MSRV check https://github.com/rust-lang/rust/issues/65262.
+
+                // HACK(nilstrieb): CURRENT_RUSTC_VERSION can return versions like 1.66.0-dev. `rustc-semver` doesn't accept
```

Comment:
@est31 might be good to followup on this and figure out what's happening here -- we probably shouldn't be returning -dev in the version, since we don't include -nightly for example.

Reply:
Yeah the `-dev` is being returned because the `CFG_VERSION` env var contains it. That env var itself is set by the bootstrap tooling (and it also adds `-nightly` postfixes FTR). I originally went with the most simple code and didn't strip the postfix because that would have involved parsing the version number (like this code is doing). I'd thought it wouldn't be much of a problem, it seems that most of the tooling could handle the postfix.

I like the `-nightly` postfix because it gives it a notion of preliminarity: attention that compiler has not been released yet, the stabilization might be reverted (like never has been multiple times, or let chains very recently). Also, sometimes that number can lie: for example right now the `cast_const` function is shown to be stabilized on `1.66.0-nightly` in the nightly docs:

![Screenshot_20220926_083953](https://user-images.githubusercontent.com/8872119/192209698-0146d7a6-6a9b-4ca7-99e5-ef968c30188a.png)

Even though it's actually been stabilized in `1.65.0`, and beta already correctly shows this:

![Screenshot_20220926_084224](https://user-images.githubusercontent.com/8872119/192210004-b7ba3640-7cb6-42f0-af92-1a208c00e97c.png)

Admittedly, this only occurs during a ~one week period during the release cycle, in this particular instance because #102051 hasn't been merged yet to master, which changes the placeholder to `1.65.0`. But I think the `-nightly` postfix helps here in this instance, makes people more cautious that it might be wrong. Or idk.

So I'm not *that* unhappy with the current postfix, but I'm willing to remove it, if it's really not wanted. #101772 has introduced the `rust_version_symbol` function, so there is a centralized place to change it, to add code not dissimilar from this one.